modules/tensor_mechanics/test/tests/crystal_plasticity/stress_update_material_based/exception_backstress.i

```diff
@@ -0,0 +1,227 @@
+[GlobalParams]
+  displacements = 'ux uy uz'
+[]
+
+[Mesh]
+  type = GeneratedMesh
+  dim = 3
+  elem_type = HEX8
+[]
+
+[AuxVariables]
+  [./pk2]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./fp_zz]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./rotout]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./e_zz]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./gss]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./slip_increment]
+   order = CONSTANT
+   family = MONOMIAL
+  [../]
+  [./backstress]
+   order = CONSTANT
+   family = MONOMIAL
+  [../]
+[]
+
+[Modules/TensorMechanics/Master/all]
+  strain = FINITE
+  add_variables = true
+  generate_output = stress_zz
+[]
+
+[AuxKernels]
+  [./fp_zz]
+    type = RankTwoAux
+    variable = fp_zz
+    rank_two_tensor = plastic_deformation_gradient
+    index_j = 2
+    index_i = 2
+    execute_on = timestep_end
+  [../]
+  [./pk2]
+   type = RankTwoAux
+   variable = pk2
+   rank_two_tensor = second_piola_kirchhoff_stress
+   index_j = 2
+   index_i = 2
+   execute_on = timestep_end
+  [../]
+  [./e_zz]
+    type = RankTwoAux
+    variable = e_zz
+    rank_two_tensor = total_lagrangian_strain
+    index_j = 2
+    index_i = 2
+    execute_on = timestep_end
+  [../]
+  [./gss]
+    type = MaterialStdVectorAux
+    variable = gss
+    property = slip_resistance
+    index = 0
+    execute_on = timestep_end
+  [../]
+  [./slip_inc]
+   type = MaterialStdVectorAux
+   variable = slip_increment
+   property = slip_increment
+   index = 0
+   execute_on = timestep_end
+  [../]
+  [./backstress]
+   type = MaterialStdVectorAux
+   variable = backstress
+   property = backstress
+   index = 0
+   execute_on = timestep_end
+  [../]
```

Comment:
Since your code is adding in the backstress variable, I would recommend having an auxkernal and auxvariable pair for each slip system, or at least one for auxkernel + auxvariable pair for each slip plane.

Reply:
resolved